thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobRunner.java

```diff
@@ -72,32 +70,45 @@ public void execute(JobExecutionContext jobExecutionContext) throws JobExecution
     } else {
       detectionJobContext.setAnomalyFunctionSpec(anomalyFunctionSpec);
 
+      DateTime monitoringWindowStartTime = detectionJobContext.getWindowStartTime();
+      DateTime monitoringWindowEndTime = detectionJobContext.getWindowEndTime();
+      boolean needCleanUpMonitoringWindow = (monitoringWindowEndTime == null);
```

Comment:
I've removed the conditional restoration. Now this method always restore the original state of JobContext at the end regardless the job is scheduled or ad-hoc one.